addon/model.js

```diff
@@ -98,33 +112,76 @@ class YesManAttributesSingletonClass {
 
 const YesManAttributes = new YesManAttributesSingletonClass();
 
-const retrieveFromCurrentState = computed('_topModel.currentState', function(key) {
-  return this._topModel._internalModel.currentState[key];
-}).readOnly();
-
 // global buffer for initial properties to work around
 //  a)  can't write to `this` before `super`
 //  b)  core_object writes properties before calling `init`; this means that no
 //      CP or setknownProperty can rely on any initialization
 let initProperites = Object.create(null);
 
+let ignoreInvalidRequestsMap = new WeakMap();
+let ignoreErrorRequestsMap = new WeakMap();
 export default class MegamorphicModel extends EmberObject {
   init(properties) {
     // Drop Ember.Object subclassing instead
     super.init(...arguments);
+    if (false) {
+      recordDataToRecordMap.set(properties._recordData, this);
+      this._recordData = properties._recordData;
+      this._invalidRequests = [];
+      this._errorRequests = [];
+      this._lastError = null;
+    }
     this._store = properties.store;
-    this._internalModel = properties._internalModel;
     this._cache = Object.create(null);
     this._schema = get(properties.store, '_schemaManager');
 
     this._topModel = this._topModel || this;
     this._parentModel = this._parentModel || null;
     this._errors = DS.Errors.create();
     this._init = true;
+    if (!false) {
+      this._internalModel = properties._internalModel;
+    }
 
     this._flushInitProperties();
   }
 
+  _setIdentifier(identifier) {
+    if (false) {
+    }
```

Comment:
probably the body of this method should be inside the then-clause here

Reply:
done